ftw/referencewidget/tests/test_relation_choice.py

```diff
@@ -0,0 +1,110 @@
+from ftw.builder import Builder
+from ftw.builder import create
+from ftw.referencewidget.tests import FunctionalTestCase
+from ftw.testbrowser import browsing
+from ftw.testbrowser.pages import statusmessages
+import json
+import transaction
+
+
+class TestRelationChoice(FunctionalTestCase):
+
+    def setUp(self):
+        super(TestRelationChoice, self).setUp()
+        self.setup_fti(additional_behaviors=[
+            'ftw.referencewidget.behaviors.IRelationChoice'])
+        self.grant('Manager')
+
+    @browsing
+    def test_relation_choice(self, browser):
+        folder = create(Builder('folder').titled(u'Some folder'))
+
+        content = create(Builder('sample content').titled(u'Sample content'))
+
+        browser.login().visit(content, view='@@edit')
+        browser.fill({'Related Choice': folder})
+        browser.find_button_by_label('Save').click()
+
+        self.assertEquals(folder, content.realtionchoice.to_object)
+
+        browser.login().visit(content, view='@@edit')
+        selected = json.loads(browser.css(
+            '.selected_items').first.attrib['data-select'])
+        self.assertEquals('/'.join(folder.getPhysicalPath()),
+                          selected[0]['path'])
+
+    @browsing
+    def test_relation_choice_with_removed_relation(self, browser):
+        folder1 = create(Builder('folder').titled(u'Some folder'))
+
+        content = create(Builder('sample content')
+                         .titled(u'Sample content')
+                         .having(realtionchoice=folder1))
+
+        self.portal.manage_delObjects([folder1.getId()])
+        transaction.commit()
+
+        self.assertIsNone(content.realtionchoice.to_object)
+
+        browser.login().visit(content, view='@@edit')
+        self.assertNotIn('data-select',
+                         browser.css('.selected_items').first.attrib)
+
+
+class TestRelationChoiceRestricted(FunctionalTestCase):
+
+    def setUp(self):
+        super(TestRelationChoiceRestricted, self).setUp()
+        self.setup_fti(additional_behaviors=[
+            'ftw.referencewidget.behaviors.IRelationChoiceRestricted'])
+        self.grant('Manager')
+
+    @browsing
+    def test_relation_choice_folder_is_not_allowed(self, browser):
+        folder = create(Builder('folder').titled(u'Some folder'))
+
+        content = create(Builder('sample content').titled(u'Sample content'))
+
+        browser.login().visit(content, view='@@edit')
+        browser.fill({'Related Choice Restricted': folder})
+        browser.find_button_by_label('Save').click()
+
+        self.assertEquals(['There were some errors.'],
+                          statusmessages.error_messages())
```

Comment:
I added a second test assertion for the message. 

Because of the widget structure I'm not able to use the ftw.testbrowser.pages.dexterity erroneous_fields method. I gonna change the structure later. 

Further this error message only appears if a user tries to do something strange, like manipulate the DOM.